quaireaux/data_structures/src/vec.cairo

```diff
@@ -0,0 +1,113 @@
+//! Vec implementation.
+//!
+//! # Example
+//! ```
+//! use quaireaux_data_structures::vec::VecTrait;
+//!
+//! // Create a new vec instance.
+//! let mut vec = VecTrait::<u128>::new();
+//! // Push some items to the vec.
+//! vec.push(1);
+//! vec.push(2);
+//! ...
+//! ```
+
+// Core lib imports
+use dict::Felt252DictTrait;
+use option::OptionTrait;
+use traits::Into;
+
+struct Vec<T> {
+    items: Felt252Dict<T>,
+    len: usize,
+}
+
+impl DestructVec<T,
+impl TDrop: Drop<T>,
+impl TFelt252DictValue: Felt252DictValue<T>> of Destruct<Vec<T>> {
+    fn destruct(self: Vec<T>) nopanic {
+        self.items.squash();
+    }
+}
+
+
+trait VecTrait<T> {
+    fn new() -> Vec<T>;
+    fn get(ref self: Vec<T>, index: usize) -> Option<T>;
+    fn at(ref self: Vec<T>, index: usize) -> T;
+    fn push(ref self: Vec<T>, value: T) -> ();
+    fn set(ref self: Vec<T>, index: usize, value: T);
+    fn len(self: @Vec<T>) -> usize;
+}
```

Comment:
IMO you should define this as such:
```rust
ait VecTrait<U, T> {
     fn new() -> U;
     fn get(ref self: U, index: usize) -> Option<T>;
     fn at(ref self: U, index: usize) -> T;
     fn push(ref self: U, value: T) -> ();
     fn set(ref self: U, index: usize, value: T);
     fn len(self: @U) -> usize;
 }
```

Reply:
J'avais pas vu ta réponse, sorry. Ba parce que le concept de ton interface 'vecteur' est pas propre au vecteur, mais juste à un conteneur à l'interface similaire. Si je veux une struct qui a la même interface mais pas le même layout en mémoire, je peux pas le faire.